### PR TITLE
bare bones process registry

### PR DESCRIPTION
**Pull request by Tuxified**

### CODE REVIEW

File: lib/horde.ex [hunk 1]
```diff
@@ -42,58 +97,134 @@ defmodule Horde do
...
+    processes = GenServer.call(state.processes_pid, {:read, AddWinsFirstWriteWinsMap})
+
+    processes_pids =
+      Enum.into(processes, MapSet.new(), fn {_name, {pid}} -> pid end)
+
+    state_processes_pids =
+      Enum.into(state.processes, MapSet.new(), fn {_name, {pid}} -> pid end)
+
+    if MapSet.difference(processes_pids, state_processes_pids) |> Enum.any?() do
+      Kernel.send(state.processes_pid, :ship_interval_or_state_to_all)
```
Comment (derekkraan) [inline lib/horde.ex]: We already do this at line 211, I think we can remove this whole bit?

File: lib/horde.ex [hunk 2]
```diff
@@ -31,9 +33,62 @@ defmodule Horde do
...
+  def lookup({:via, _, {horde, name}}), do: lookup(horde, name)
+
+  def lookup(horde, name) do
+    case GenServer.call(horde, {:lookup, name}) do
+      {:ok, pid} -> pid
+      _ -> :undefined
+    end
+  end
+
+  def start_handoff(name) do
```
Comment (derekkraan) [inline lib/horde.ex]: This can be removed

File: lib/horde.ex [hunk 3]
```diff
@@ -42,58 +97,134 @@ defmodule Horde do
...
+      Enum.into(state.processes, MapSet.new(), fn {_name, {pid}} -> pid end)
+
+    if MapSet.difference(processes_pids, state_processes_pids) |> Enum.any?() do
+      Kernel.send(state.processes_pid, :ship_interval_or_state_to_all)
+    end
+
+    {:noreply, %{state | processes: processes}}
+  end
+
+    def handle_info(:members_updated, state) do
```
Comment (derekkraan) [inline lib/horde.ex]: indentation (run mix format?)

File: lib/horde.ex [hunk 4]
```diff
@@ -161,35 +160,29 @@ defmodule Horde do
...
-      Enum.into(processes, MapSet.new(), fn {_name, {pid}} -> pid end)
+    processes_pids = Enum.into(processes, MapSet.new(), fn {_name, {pid}} -> pid end)
 
-    state_processes_pids =
-      Enum.into(state.processes, MapSet.new(), fn {_name, {pid}} -> pid end)
-
-    if MapSet.difference(processes_pids, state_processes_pids) |> Enum.any?() do
-      Kernel.send(state.processes_pid, :ship_interval_or_state_to_all)
-    end
+    state_processes_pids = Enum.into(state.processes, MapSet.new(), fn {_name, {pid}} -> pid end)
```
[the remark this answers is not in the document]
Comment (Tuxified) [reply lib/horde.ex]: Yes, I also noticed that :)

**Pull request merged**